lib/config/config-initializer.js

```diff
@@ -342,13 +342,18 @@ function promptUser(callback) {
             default: "JavaScript",
             choices: ["JavaScript", "YAML", "JSON"],
             when: function(answers) {
-                return (answers.source === "guide" || answers.source === "auto");
+                return ((answers.source === "guide" && answers.packageJsonExists) || answers.source === "auto");
             }
         }
     ], function(earlyAnswers) {
 
         // early exit if you are using a style guide
         if (earlyAnswers.source === "guide") {
+            if (!earlyAnswers.packageJsonExists) {
```

Comment:
Didn't add tests for this because of https://github.com/eslint/eslint/pull/6478/files#diff-da527c6839c0e6e4d343b5c77fcdd27cR293.

By logging and returning below it exits with a nice message and doesn't show the stack trace